src/main/kotlin/com/ampnet/payoutservice/util/MerkleTree.kt

```diff
@@ -29,35 +30,46 @@ class MerkleTree(nodes: List<AccountBalance>, val hashFn: HashFunction) {
             override val hash: Hash,
             val depth: Int
         ) : PathNode
+
+        @JsonSerialize(using = PathSegmentJsonSerializer::class)
+        data class PathSegment(val hash: Hash, val isLeft: Boolean)
```

Comment:
Could you please explain why this class, more precisely `isLeft`, is needed? We can do it over a call but it would also be nice to have it documented.

Reply:
So let's say we have a simple Merkle tree with four nodes, their hashes are: A, B, C and D. Their parent nodes have hashes Hash(A + B) and Hash(C + D). The Merkle root hash is Hash(Hash(A + B) + Hash(C + D)). Now let's say we want to path to C node. The path would be: D, Hash(A + B). However, to verify that the path is correct we need to know for each segment if it comes before or after the C node hash (or any other hash in the chain). So, D part of the path would have `isLeft` set to false because to calculate the parent hash, you do Hash(C + D), NOT Hash (D + C) - these two hashes are not the same. Hash(A + B) in the path would have `isLeft` set to true because you calculate root hash as Hash(Hash(A + B) + Hash(C + D)) - so Hash(A + B) is "left" of Hash(C + D).

TLDR: `isLeft` is used to determine where should the sibling hash be concatenated when calculating parent hash: left or right from the current node hash.